tests/integration/test_ndim.py

```diff
@@ -34,6 +34,23 @@ def test_python_values(input):
     assert np.ndim(input) == num.ndim(input)
 
 
+def test_ndarray_none():
+    inp = None
+    assert np.ndim(inp) == num.ndim(inp)
+
+
+@pytest.mark.parametrize("input", ([], (), (()), ((), ()), [[]], [[], []]))
+def test_ndarray_empty(input):
+    assert np.ndim(input) == num.ndim(input)
+
+
+@pytest.mark.parametrize(
+    "input", ([[0], [1, 2], [3, 4, 5]], [[1, 2], [3.3, 4.4]])
```

Comment:
The extra parens here do nothing, either they or the outer list can be removed:
```
In [2]: [x for x in ([[1], [2]])] == [x for x in ([1], [2])]
Out[2]: True
```

I suggest removing the outer list (making the parens actually do something by being a tuple):
```suggestion
    "input", ([0], [1, 2], [3, 4, 5]], [[1, 2], [3.3, 4.4])
```

Reply:
fix with tuple:  "input", (([0], [1, 2], [3, 4, 5]), ([1, 2], [3.3, 4.4])).  I intend it to be 2 parameters.

Reply:
ah, I missed the inner brackets, thanks for changing I do find this version much easier to parse